examples/example_vulkan.cpp

```diff
@@ -45,9 +47,177 @@ void testNonEngineIntergratedVulkan()
+        // TODO: should probably not use GetSupportingQueueIndex.. IsQueueFamilySupportingFlags might be better to use
+
+        // TODO: this following part makes no sense.. rewrite when we know how we want to deal with this
+        if (isEverythingSupported) {
+            std::cout << "\t'VkQueueFamilyProperties' at index " << allIndex << " supports all functionalities." << std::endl;
+            queueInfos.push_back(QueueInfo({ gfxIndex }));
+            for (uint32_t i = 0; i < queueFamilies[gfxIndex].queueCount; ++i) {
+                queueInfos[0].priorities.push_back(1.0f);
+            }
+            selectedDevice = device;
+            break;
+        }
+
+        if (isGfxPipelineSupported) {
+            std::cout << "\t'VkQueueFamilyProperties' at index " << gfxIndex << " supports praphics pipelines." << std::endl;
+            queueInfos.push_back(QueueInfo({ gfxIndex }));
+            for (uint32_t i = 0; i < queueFamilies[gfxIndex].queueCount; ++i) {
+                queueInfos[0].priorities.push_back(1.0f);
+            }
+            selectedDevice = device;
+            break;
+        }
+
+        if (!isEverythingSupported && !isGfxPipelineSupported) {
+            std::cerr << "WARNING: None of the VkQueueFamilyProperties has the required functionalites." << std::endl;
+        }
+    }
+
+    if (selectedDevice == VK_NULL_HANDLE) {
+        std::cerr << "ERROR: No device has been picked.. this is most likely a logical/code issue.. fix it" << std::endl;
+        return;
+    }
+
+    std::vector<VkDeviceQueueCreateInfo> queueCreateInfos;
+    for (const QueueInfo& info : queueInfos) {
+        // TODO: Should be done in initializer list.. but better for clarity now
+        VkDeviceQueueCreateInfo ci;
+        ci.sType = VK_STRUCTURE_TYPE_DEVICE_QUEUE_CREATE_INFO;
+        ci.pNext = nullptr;
+        ci.flags = 0;
+        ci.queueFamilyIndex = info.familyIndex;
+        ci.queueCount = static_cast<uint32_t>(info.priorities.size());
+        ci.pQueuePriorities = info.priorities.size() > 0 ? info.priorities.data() : nullptr;
+        queueCreateInfos.push_back(ci);
     }
+
+    // Logical device
+    // TODO: Should be done in initializer list.. but better for clarity now
+    VkDeviceCreateInfo deviceCreateInfo;
+    deviceCreateInfo.sType = VK_STRUCTURE_TYPE_DEVICE_CREATE_INFO;
+    deviceCreateInfo.pNext = nullptr;
+    deviceCreateInfo.flags = 0;
+    deviceCreateInfo.queueCreateInfoCount = static_cast<uint32_t>(queueCreateInfos.size());
+    deviceCreateInfo.pQueueCreateInfos = queueCreateInfos.size() > 0 ? queueCreateInfos.data() : nullptr;
+    deviceCreateInfo.enabledLayerCount = 0;
+    deviceCreateInfo.ppEnabledLayerNames = nullptr;
+    deviceCreateInfo.enabledExtensionCount = static_cast<uint32_t>(desiredDeviceExtensions.size());
+    deviceCreateInfo.ppEnabledExtensionNames = desiredDeviceExtensions.size() > 0 ? &desiredDeviceExtensions[0] : nullptr;
```

Comment:
can we pass all of this stuff as some metadata/config later?

Reply:
Yeah, something like that